analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/VerifierBuilder.cs

```diff
@@ -74,7 +74,10 @@ this with
             this with { References = References.Concat(references).ToImmutableArray() };
 
         public VerifierBuilder AddSnippet(string snippet, string fileName = null) =>
-            this with { Snippets = Snippets.Append(new(snippet, fileName)).ToImmutableArray() };
+            this with {
+                Snippets = Snippets.Append(new(snippet, fileName)).ToImmutableArray(),
```

Comment:
This is a bit tricky to predict performancewise :-)

`Append` is an extension method from `System.Linq` which is O(1) (see [here](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/AppendPrepend.cs#L89)), and that in principle is good. 

However, `ToImmutableArray` is in general O(n), when it takes an `IEnumerable<TSource>`:
```cs
public static ImmutableArray<TSource> ToImmutableArray<TSource>(this IEnumerable<TSource> items)
{
    if (items is ImmutableArray<TSource>)
    {
        return (ImmutableArray<TSource>)items;
    }

    return CreateRange(items);
}
```
While a special case is defined for `ImmutableArray<TSource>`, the Append before returns a `AppendPrepend1Iterator`, which is not an `ImmutableArray<TSource>`.
 
It is true that the case of an `IImmutableArray` is handled specifically and O(1) in some cases, such as [this](https://github.com/dotnet/runtime/blob/a7d4b4b4761b68c85e27d9c4c8dacb86c285ddc6/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs#L144).
However, if I have read the code correctly, that doesn't seem to apply to the overload of `Create` that is used by this call.

So, I would propose to use `Snippets.Add(new(snippet, fileName))` ([ref](https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablearray-1.add?view=net-7.0)), which returns a new `ImmutableArray<Snippet>` and should be O(log(n)) (at most?).